src/components/Popup.js

```diff
@@ -26,26 +20,14 @@ export default class Popup {
     }
   };
 
-  uploadingInfo(isloading) {
-    if (isloading) {
-      this._submitButtonElement.classList.add("modal_button_saving");
-      this._submitButtonElement.textContent = "Saving...";
-    } else {
-      this._submitButtonElement.classList.remove("modal_button_saving");
-      this._submitButtonElement.textContent = "Save";
-    }
-  }
-
   _handleClickOutside = (evt) => {
     if (evt.target.classList.contains("modal_open")) {
       this.close();
     }
   };
 
   _handleCloseButtons = () => {
-    closeButtons.forEach((button) => {
-      button.addEventListener("click", () => this.close());
-    });
+    this._closeButton.addEventListener("click", () => this.close());
```

Comment:
You can't add clicks to `document` inside which you add a click to the button. (in line 37 and 43) otherwise you add listeners again and again to the same elements. This causes a **`memory leak`**



You need to add the click to the button and remove the click from the **button**


So, here will be only ` this.close()`


```js

  _handleCloseButtons = () => {
    this.close()
  };

```


And in line 37 and 43 you need to add and remove the listener from `this._closeButton`